### PR TITLE
[Hieu] Chat Home

### PR DESCRIPTION
**Pull request by HieuJC**

### CODE REVIEW

File: app/src/main/java/com/hieuminh/chatapp/ui/adapter/HomePagerAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import androidx.fragment.app.FragmentStatePagerAdapter
+import com.hieuminh.chatapp.constant.Constant
+import com.hieuminh.chatapp.ui.error.ErrorHomeFragment
+import com.hieuminh.chatapp.ui.home.ChatHomeFragment
+import com.hieuminh.chatapp.ui.home.SettingHomeFragment
+
+class HomePagerAdapter(fragmentManager: FragmentManager) : FragmentStatePagerAdapter(fragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
+
+    override fun getCount(): Int {
+        return 2
```
Comment (chuongdevJC) [inline app/src/main/java/com/hieuminh/chatapp/ui/adapter/HomePagerAdapter.kt]: create a fragments: List<Fragment> to handle count or show fragment index

File: app/src/main/java/com/hieuminh/chatapp/presenter/contract/HomeContact.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+interface HomeContact {
+
+    interface Presenter : BasePresenter {
+        fun init()
+    }
+
+    interface View {
+        fun initTabLayout()
+        fun initViewPager()
+    }
```
Comment (vuphu-jc) [inline app/src/main/java/com/hieuminh/chatapp/presenter/contract/HomeContact.kt]: call init() at HomeContract.View

File: app/src/main/java/com/hieuminh/chatapp/ui/adapter/HomePagerAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    override fun getItem(position: Int): Fragment {
+        return when(position) {
+            Constant.LayoutIndex.HOME -> ChatHomeFragment()
+            Constant.LayoutIndex.SETTING -> SettingHomeFragment()
+            else -> {
+                ErrorHomeFragment()
+            }
```
Comment (vuphu-jc) [inline app/src/main/java/com/hieuminh/chatapp/ui/adapter/HomePagerAdapter.kt]:
```suggestion
            else -> ErrorHomeFragment()
```

File: app/src/main/java/com/hieuminh/chatapp/presenter/contract/HomeContact.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.hieuminh.chatapp.presenter.contract
+
+import com.hieuminh.chatapp.presenter.BasePresenter
+
+interface HomeContact {
```
Comment (AnhJobChat) [inline app/src/main/java/com/hieuminh/chatapp/presenter/contract/HomeContact.kt]:
```suggestion
interface HomeContract {
```

File: app/src/main/java/com/hieuminh/chatapp/ui/adapter/HomePagerAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        }
+    }
+
+    override fun getPageTitle(position: Int): CharSequence {
+        return when(position) {
+            Constant.LayoutIndex.HOME -> "Chat"
+            Constant.LayoutIndex.SETTING -> "Setting"
```
Comment (AnhJobChat) [inline app/src/main/java/com/hieuminh/chatapp/ui/adapter/HomePagerAdapter.kt]: get value from strings

File: app/src/main/java/com/hieuminh/chatapp/presenter/contract/HomeContract.kt [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import com.hieuminh.chatapp.presenter.BasePresenter
+
+interface HomeContract {
+
+    interface Presenter : BasePresenter {
+    }
+
+    interface View {
+        fun initTabLayout()
+        fun initViewPager()
```
Comment (vuphu-jc) [inline app/src/main/java/com/hieuminh/chatapp/presenter/contract/HomeContract.kt]: remove

File: app/src/main/java/com/hieuminh/chatapp/ui/home/HomeFragment.kt [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        super.onViewCreated(view, savedInstanceState)
+        init()
+    }
+
+    override fun onDestroy() {
+        mHomePresenter.onDestroy()
+        super.onDestroy()
+    }
+
+    override fun initViewPager() {
```
Comment (vuphu-jc) [inline app/src/main/java/com/hieuminh/chatapp/ui/home/HomeFragment.kt]:
```suggestion
   private fun initViewPager() {
```

**Review by AnhJobChat — APPROVED** [review]

File: app/src/main/java/com/hieuminh/chatapp/ui/adapter/HomePagerAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+    private val fragments: List<Fragment> = arrayListOf(ChatHomeFragment(), SettingHomeFragment())
+
+    override fun getCount(): Int {
+        return fragments.size
+    }
+
+    override fun getItem(position: Int): Fragment {
+        return when (position) {
+            TabHomeType.HOME.index -> fragments[TabHomeType.HOME.index]
```
Comment (chuongdevJC) [inline app/src/main/java/com/hieuminh/chatapp/ui/adapter/HomePagerAdapter.kt]:
```suggestion
            TabHomeType.HOME.index -> fragments[position]
```

**Review by chuongdevJC — APPROVED** [review]

**Review by vuphu-jc — APPROVED** [review]

**Pull request merged**